src/main/java/individual/freshplace/util/payment/KakaoPay.java

```diff
@@ -0,0 +1,134 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+public class KakaoPay {
+
+    private static final long ORDER_ID_MIN_VALUE = 1;
+    private static final long ORDER_ID_MAX_VALUE = 10000000000L;
+
+    private final RestTemplate restTemplate;
+    private final KakaoPayProperties kakaoPayProperties;
+
+    public PayView getKAKAOPayReadyResponse(final String memberId, final List<OrderItem> orderItems) {
+
+        HttpHeaders httpHeaders = setHeaders();
+
+        MultiValueMap<String, String> params = setParamsReady(memberId, orderItems.get(0).getItemName() + " 외" + (orderItems.size() - 1) + "건"
+                , orderItems.stream().map(OrderItem::getItemCount).reduce(0L, (a, b) -> a + b)
+                , orderItems.stream().map(OrderItem::getTotalPrice).reduce(0L, (a, b) -> a + b));
+        setParamsCommon(memberId, params);
+
+        HttpEntity<MultiValueMap<String, String>> body = new HttpEntity<>(params, httpHeaders);
+
+        KakaoPayReadyResponse kakaoPayReadyResponse;
+        String paymentReadyRequestUrl = kakaoPayProperties.getHost() + kakaoPayProperties.getReadyPath();
+
+        try {
+            kakaoPayReadyResponse = restTemplate.postForObject(new URI(paymentReadyRequestUrl), body, KakaoPayReadyResponse.class);
+        } catch (URISyntaxException e) {
+            throw new UriException(ErrorCode.NON_FOUND, paymentReadyRequestUrl);
+        }
+
+        return PayView.of(kakaoPayReadyResponse.getTid(), params.getFirst(KakaoPayProperties.ORDER_ID_KEY),
+                memberId, kakaoPayReadyResponse.getNextRedirectPcUrl());
+    }
+
+    public KakaoPayApprovalResponse getKAKAOPaymentInformation(final String pgToken, final PayView payView) {
+
+        HttpHeaders httpHeaders = setHeaders();
+
+        MultiValueMap<String, String> params = setParamsApprove(pgToken, payView);
+        setParamsCommon(payView.getUserId(), params);
+
+        HttpEntity<MultiValueMap<String, String>> body = new HttpEntity<>(params, httpHeaders);
+
+        KakaoPayApprovalResponse kakaoPayApprovalResponse;
+        String paymentApproveRequestUrl = kakaoPayProperties.getHost() + kakaoPayProperties.getApprovePath();
+
+        try {
+            kakaoPayApprovalResponse = restTemplate.postForObject(new URI(paymentApproveRequestUrl), body, KakaoPayApprovalResponse.class);
+        } catch (URISyntaxException e) {
+            throw new UriException(ErrorCode.NON_FOUND, paymentApproveRequestUrl);
+        }
+
+        return kakaoPayApprovalResponse;
```

Comment:
변수는 선언시 꼭 초기화해주세요. 초기화를 하지 않고 다른곳에서 초기화할 경우 NPE발생의 가능성이 높아지고, 어느곳에서 데이터가 들어가는지 추적하기 어려워집니다.

또한 try-catch 내부에서만 사용한다면 해당 블록 내부에 선언해서 변수를 외부로 노출하지 말아주세요.
```suggestion
        

        try {
         KakaoPayApprovalResponse   kakaoPayApprovalResponse = restTemplate.postForObject(new URI(paymentApproveRequestUrl), body, KakaoPayApprovalResponse.class);
         return kakaoPayApprovalResponse;
        } catch (URISyntaxException e) {
            throw new UriException(ErrorCode.NON_FOUND, paymentApproveRequestUrl);
        }

        
```

Reply:
> 

try catch 내부에서 return 하도록 하겠습니다.